pkg/completion/completion.go

```diff
@@ -67,33 +68,47 @@ func (wg *WaitGroup) AddCompletion() *Completion {
 // Wait blocks until all completions added by calling AddCompletion are
 // completed, or the context is canceled, whichever happens first.
 // Returns the context's error if it is cancelled, nil otherwise.
+// No callbacks of the completions in this wait group will be called after
+// this returns.
+// Returns the error value of one of the completions, if available, or the
+// error value of the WaitGroup otherwise.
 func (wg *WaitGroup) Wait() error {
 	wg.counterLocker.Lock()
 	defer wg.counterLocker.Unlock()
 
+	var err error
 Loop:
 	for i, comp := range wg.pendingCompletions {
 		select {
 		case <-comp.Completed():
+			err = comp.Error
```

Comment:
You shouldn't overwrite any previous error if `comp.Error` is `nil`.
Also, to be consistent with the other cases below, I would only set `err` if it is currently `nil`.
```
if err == nil && comp.Error != nil {
    err = comp.Error
}
```

Reply:
changed